src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,11 +1047,28 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static SyntaxKind[] _lazyContextualKeywordKindsArray;
```

Comment:
If you're going to cache the results of this operation, i would recommend just typing it as an IReadOnlyList.  THen, in a static constructor, do the computation using an ```ArrayBuilder<SyntaxKind>```.